### DISCUSSION TITLE
When streaming on OBS, Pipewire screen/window capture stops working.

### DISCUSSION THREAD

**Discussion by yuurierusan** [prasanthrangan/hyprdots#711] [Q&A] [answered]
Good Afternoon y'all, 

System Info & OBS log file:
Arch Linux 6.6.8 / Hyprland
AMD 5800X3D / AMD RX6950XT
OBS via flatpak
https://obsproject.com/logs/MXdrIL75xKFyPNom

I am having an issue with window/screen capture using Pipewire. On start after i select a window or screen it works fine. If I switch sources like going from full screen in one scene to PIP(picture in picture) with the chat next to it and my face cam, the game/window being recorded freezes. Then when i go to properties to set it again it just goes black. At that point the stream has to end because it either crashes after 30 sec or I have to restart because I cant get the screen to come back up. 

Based on the log it looks like it connects to the picture then if i switch scenes it gets a "no target node available" error. 

So far I've tried: 
-Reboot & Logout (temporarily fixes it but eventually happens on scene switch)
-Uninstalling & Reinstalling (temporarily fixes it but eventually happens on scene switch)
-Running in safe mode (temporarily fixes it but eventually happens on scene switch)
-Updating dependencies (e.g Pipewire, xdg-desktop-portal, etc)
-Updating system via sudo pacman -Syu (although nothing happened, guess it was already up to date)
-I installed Flatseal cause another forum told me it was necessary or OBS wouldn't have the permissions to run correctly, but half the stuff in flatseal permission i don't understand so I'm not sure what to change.
-Lastly, i went to OBS devs for help, in the forum they said ->
"complain to your desktop portal implementation since it seems they report the error. We dont retry on errors at the moment.
as far as im aware all errors are basically fatal, so retrying wouldnt help when they are reported."

Obviously im not complaining to you guys cause im not even sure this is actually a hyprland issue but any help is appreciated, thanks in advance for any advice!

**Comment by kRHYME7** [comment]
Halu! Do you have https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-hyprland/ installed?

**Reply by yuurierusan** [reply]
Hello, Yes, i have xdg-desktop-portal-hyprland 1.2.6-1 & xdg-desktop-portal-wlr 0.7.0-2

**Reply by ghost** [reply]
Oh, do not mix xdph and xdpw together as it causes issues. Uninstall xdpw and it might fix hopefully!

**Reply by yuurierusan** [reply]
okay i uninstalled the xdpw and it worked for a few scene switches then it froze again turned black and gave this error in the logs
![image](https://github.com/prasanthrangan/hyprdots/assets/121768011/0cf79141-b0ca-4ce6-b964-6743a14550d2)

**Reply by yuurierusan** [reply]
i tried going back to my kde plasma desktop with the sddm and screen capture is working as intended on x11 and wayland, still havent figured out whats different between the 2 or what dependencies are working one place versus the other but ill just use kde for now, although i dont particularly like it lol

**Reply by ghost** [reply, marked as answer]
Looks like a new xdph version just released with screenshare fixes, might be helpful: https://github.com/hyprwm/xdg-desktop-portal-hyprland/releases/tag/v1.3.0

Other than that i'm not entirely sure, because i'm not having issues and i don't really know what to do with that error.

**Reply by yuurierusan** [reply]
I was trying to build that and I just get a error during compile of 18-33, maybe i'm building it wrong. But if not ill probably end up making a new issue on the their github.

**Reply by yuurierusan** [reply]
okay i finally got it built with their help and the error has been partially resolved, thank you guys @Skerse  @kRHYME7
